python/lsst/ts/cbp/component.py

```diff
@@ -261,52 +247,48 @@ def change_focus(self, position: int):
         if position < 0 or position > 13000:
             raise ValueError("New focus value exceeds focus limit.")
         else:
-            self.socket.sendall("new_foc={0:f}\r".format(position).encode('ascii'))
-            reply = self.socket.recv(128).decode('ascii', 'ignore')
-            self.log.debug(reply)
+            await self.send_command(f"new_foc={position}")
 
-    def get_mask(self):
-        """This gets the current mask value from the encoder which is converted into the name of the mask.
+    async def get_mask(self):
+        """This gets the current mask value from the encoder which is converted
+        into the name of the mask.
 
         Returns
         -------
         None
 
         """
-        self.socket.sendall("msk=?\r".encode('ascii'))
-        self.mask = self.mask_id_dictionary[float(self.parse_reply())].name
+        mask = str(int(float(await self.send_command("msk=?"))))
```

Comment:
This code worries me. If `masks.__dict__.get(mask)` returns `None` the code will raise an exception. How likely is that to happen?

If the encoder has typical encoder resolution then that looks like the _most likely_ outcome.

If the encoder has exactly N positions, where N is the number of masks, then the code is fine, but some explanation to the reader would be very helpful. And what happens if the mask rotator is lost or between mask positions?

Reply:
If the mask encoder is off place, then it returns "9.0" Unknown mask. Otherwise there are 5 mask slots.

Reply:
Thanks. Please add a brief code comment to that effect. It's a pity that the real controller returns a float value for integer data.

Reply:
We can change this if need be... but not for awhile. No one (including me) is allowed in the lab at the moment.